src/lumigo_tracer/wrappers/http/sync_http_wrappers.py

```diff
@@ -245,10 +246,15 @@ def _headers_reminder_wrapper(func, instance, args, kwargs):
     Remember the headers helps us to improve performances on requests that use this flow.
     """
     with lumigo_safe_execute("add hooked data"):
+        headers = safe_get_list(args, HEADERS_ARG_INDEX_REQUEST) or kwargs.get("headers") or {}
+        if Configuration.propagate_w3c:
+            SpansContainer.get_span().add_w3c_trace_propagator(headers)
+            kwargs["headers"] = headers
+            args = args[:HEADERS_ARG_INDEX_REQUEST]
```

Comment:
Why removing the args after the headers?

Reply:
This is deep!
This is the signature of the wrapped function: 
<img width="462" alt="image" src="https://user-images.githubusercontent.com/47108628/187879887-45fb082a-468c-4d97-9cee-44b13766921b.png">
We want to handle both cases: whether the headers was given as args or as kwargs. In the following lines, we use the function *differently*, and move the headers as kwargs!
Therefore, if the users has passed `headers` to `HTTPConnection.request` with args - we trim it. If he passed it as kwarg, we do here nothing, and just replacing the value in the following line

Reply:
This is deep!
This is the signature of the wrapped function: 
<img width="462" alt="image" src="https://user-images.githubusercontent.com/47108628/187879887-45fb082a-468c-4d97-9cee-44b13766921b.png">
We want to handle both cases: whether the headers was given as args or as kwargs. In the following lines, we use the function *differently*, and move the headers as kwargs!
Therefore, if the users has passed `headers` to `HTTPConnection.request` with args - we trim it. If he passed it as kwarg, we do here nothing, and just replacing the value in the following line